includes/api/episodes/related_episodes.php

```diff
@@ -0,0 +1,177 @@
+
+        $isFilter = true;
+        $filter = $request->get_param('status');
+        if (!$filter || $filter != 'draft') {
+            $isFilter = false;
+        }
+
+        $episode = Episode::find_by_id($id);
+
+        if (!$episode) {
+            return new \Podlove\Api\Error\NotFound();
+        }
+
+        $relations = EpisodeRelation::find_all_by_where('left_episode_id = '.$episode->id );
+
+        $results = array_map(function($relation) use ($isFilter) {
+            $related_id = $relation->right_episode_id;
+            $related_episode = Episode::find_by_id($related_id);
+            if ($isFilter || $related_episode->is_published())
+                return $relation->right_episode_id;
+        }, $relations);
+        // Delete the invalid entries
+        $results = array_filter($results);
+
+        return new \Podlove\Api\Response\OkResponse([
+            '_version' => 'v2',
+            'releatedEpisodes' => $results
+        ]);
+    }
+
+    public function get_item_permissions_check($request)
+    {
+        $filter = $request->get_param('status');
+        if ($filter) {
+            if ($filter == 'draft') {
+                if (!current_user_can('edit_posts')) {
+                    return new \Podlove\Api\Error\ForbiddenAccess();
+                }
+            }
+        }
+        return true;
+    }
+
+    public function update_item($request)
+    {
+        $id = $request->get_param('id');
+        if (!$id) {
+            return;
+        }
+
+        $episode = Episode::find_by_id($id);
+
+        if (!$episode) {
+            return new \Podlove\Api\Error\NotFound();
+        }
+
+        if (isset($request['related'])) {
+            if (is_array($request['related'])) {
+                foreach($request['related'] as $related_id) {
+                    $this->create_episode_relation($id, $related_id);
+                }
+            }
+            else {
+                $related_id = $request['related'];
+                $this->create_episode_relation($id, $related_id);
+            }
+        }
+
+        return new \Podlove\Api\Response\OkResponse([
+            'status' => 'ok'
+        ]);
+    }
+    
+    public function update_item_permissions_check($request)
+    {
+        if (!current_user_can('edit_posts')) {
+            return new \Podlove\Api\Error\ForbiddenAccess();
+        }
+
+        return true;
+    }
+
+    public function delete_item($request)
+    {
+        $id = $request->get_param('id');
+        if (!$id) {
+            return;
+        }
+
+        $episode = Episode::find_by_id($id);
+
+        if (!$episode) {
+            return new \Podlove\Api\Error\NotFound();
+        }
+
+        $relations = EpisodeRelation::find_all_by_where('left_episode_id = '.$episode->id );
+
+        foreach($relations as $relation) {
+            $relation->delete();
+        }
+
+        return new \Podlove\Api\Response\OkResponse([
+            'status' => 'ok'
+        ]);
+    }
```

Comment:
Das löscht nicht ein, sondern alle Einträge für die Episode, oder? Sollte dann vermutlich eher `delete_items` heißen?

Ich glaube in der alten UI werden bei jedem Speichern alle relations entfernt und dann wieder alle hinzugefügt. Für die neue UI könnte es aber schon sinnvoll sein, die Möglichkeit zu haben, auch einzelne relations per id zu löschen.